presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.query.QueryResult;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Response;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.google.common.base.Preconditions.checkState;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.lang.System.nanoTime;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.fail;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private ExecutorService executor;
+    private QueryCanceller queryCanceller;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    private String serverAddress;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        executor = newFixedThreadPool(2);
+        queryCanceller = new QueryCanceller(serverAddress);
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+    {
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+        if (queryCanceller != null) {
+            queryCanceller.close();
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
+    public void cancelCreateTableTest()
+            throws Exception
+    {
+        String tableName = format("cancel_createtable_%s", nanoTime());
+        String sql = format("CREATE TABLE %s AS SELECT * FROM tpch.sf1.lineitem", tableName);
+
+        runAndThenCancelQuery(sql);
+        assertThat(() -> query(format("SELECT * from %s", tableName)))
+                .failsWithMessage(format("Table hive.default.%s does not exist", tableName));
+    }
+
+    @Test(groups = {CANCEL_QUERY}, timeOut = 60 * 1000L)
+    public void cancelInsertIntoTest()
```

Comment:
test method usually begins with `test` (not ends with it).

Reply:
I even think that is not needed. Lets remove it.